src/main/resources/category/java/common_std.xml

```diff
@@ -87,4 +87,76 @@ public class StringStuff {
         </example>
     </rule>
 
+
+    <rule name="LimitStatementsInLambdas"
+          language="java"
+          message="Avoid many statements in lambda expressions."
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          externalInfoUrl="${doc_root}/JavaCodeQuality.md#isr03">
+        <description>Problem: lambda expressions with many statements are hard to understand and maintain.
+            Solution: extract the lambda expression code block into one or more well-named separate method(s).
+        </description>
+        <priority>3</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,brain-overload,replaces-sonar-rule" type="String" description="classification"/>
+            <property name="param-max" value="5" type="String" description="Maximum number of allowed non-setter statements"/>
+            <property name="xpath">
+                <value><![CDATA[
+(: number of statements should be limited, setters are ignored. Chained methods are one statement :)
+//LambdaExpression/Block[count(.//(
+  Block|
+  BreakStatement|
```

Comment:
Ik heb even gezocht, en het kan wel! 
//*[pmd-java:nodeIs('Statement')]
Hiermee krijg je alle nodes die de interface ASTStatement implementeren, zie:
https://docs.pmd-code.org/apidocs/pmd-java/7.1.0/net/sourceforge/pmd/lang/java/ast/ASTStatement.html

Reply:
Great!  I have removed the EmptyStatement from the list, so I will make an exclude for that next to the nodeIs() construct.